cmd/build-uki.go

```diff
@@ -145,8 +145,9 @@ func NewBuildUKICmd() *cobra.Command {
 	c.Flags().StringP("boot-branding", "", "Kairos", "Boot title branding")
 	c.Flags().BoolP("include-version-in-config", "", false, "Include the OS version in the .config file")
 	c.Flags().BoolP("include-cmdline-in-config", "", false, "Include the cmdline in the .config file. Only the extra values are included.")
-	c.Flags().StringSliceP("extra-cmdline", "c", []string{}, "Add extra efi files with this cmdline. This creates a base efi with the default cmdline and extra efi files with the default+provided cmdline.")
-	c.Flags().StringP("extend-cmdline", "x", "", "Extend the default cmdline with this parameters. This creates a single efi entry with the default+provided cmdline.")
+	c.Flags().StringSliceP("extra-cmdline", "c", []string{}, "Add extra efi files with this cmdline for the default 'norole' artifacts. This creates efi files with the default cmdline and extra efi files with the default+provided cmdline.")
+	c.Flags().StringP("extend-cmdline", "x", "", "Extend the default cmdline for the default 'norole' artifacts. This creates efi files with the default+provided cmdline.")
+	c.Flags().StringSliceP("single-efi-cmdline", "s", []string{}, "Add one extra efi file with the default+provided cmdline. The syntax is '--single-efi-cmdline \"My Entry: cmdline,options,here\"'. The boot entry name is the text under which it appears in systemd-boot menu.")
```

Comment:
I think the documentation is still a bit confusing, maybe something like this:

- extra-cmdline: creates an extra efi file (besides norole) with the default + provided cmdline. This extra file will be multiplied for each role (active, passive, etc)
- extend-cmdline: extends the norole.efi with the default + provided cmdline
- single-efi-cmdline: creates a my_entry.efi with the default + provided cmdline

but maybe it's just going to be tricky to explain anyhow so not a strong opinion this one

Reply:
Your first bullet is also not 100% accurate. I think the extra lines are also "norole" but with additional cmdline options :D . It's a mess. I can change it but it's not going to be clear anyway.

I was planning to write some docs (in the kairos docs? enki repository README.md?) with some examples and use cases for each one.
Hopefully that would make it clearer.

Reply:
https://github.com/kairos-io/kairos-docs/pull/182